storage/store.go

```diff
@@ -253,8 +260,48 @@ func (si *storeRangeIterator) Reset() {
 	si.index = 0
 }
 
-// StoreConfig contains various parameters of a Store.
-type StoreConfig struct {
+// A Store maintains a map of ranges by start key. A Store corresponds
+// to one physical device.
+type Store struct {
+	*StoreFinder
+
+	Ident          proto.StoreIdent
+	ctx            StoreContext
```

Comment:
I'd just embed the StoreContext anonymously so you don't need to add `.ctx` everywhere.